stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/internal/Linkify.kt

```diff
@@ -0,0 +1,212 @@
+        val links: ArrayList<LinkSpec> = ArrayList<LinkSpec>()
+        gatherLinks(
+            links, text, PatternsCompat.AUTOLINK_WEB_URL, arrayOf("http://", "https://", "rtsp://"),
+            Linkify.sUrlMatchFilter, null
+        )
+
+        pruneOverlaps(links, text)
+
+        if (links.isEmpty()) return false
+
+        for (link in links) {
+            if (link.markwonAddedSpan == null) {
+                applyLink(link.url!!, link.start, link.end, text)
+            }
+        }
+        return true
+    }
+
+    private fun addLinkMovementMethod(t: TextView) {
+        val m = t.movementMethod
+        if (m !is LinkMovementMethod) {
+            if (t.linksClickable) {
+                t.movementMethod = LinkMovementMethod.getInstance()
+            }
+        }
+    }
+
+    private fun applyLink(
+        url: String,
+        start: Int,
+        end: Int,
+        text: Spannable,
+    ) {
+        val urlSpanFactory = DEFAULT_SPAN_FACTORY
+        val span = urlSpanFactory(url)
+        text.setSpan(span, start, end, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+    }
+
+    private fun makeUrl(
+        url: String,
+        prefixes: Array<String>,
+        matcher: Matcher,
+        filter: Linkify.TransformFilter?,
+    ): String {
+        var url = url
+        if (filter != null) {
+            url = filter.transformUrl(matcher, url)
+        }
+        var hasPrefix = false
+        for (i in prefixes.indices) {
+            if (url.regionMatches(0, prefixes[i], 0, prefixes[i].length, ignoreCase = true)) {
+                hasPrefix = true
+
+                // Fix capitalization if necessary
+                if (!url.regionMatches(0, prefixes[i], 0, prefixes[i].length, ignoreCase = false)) {
+                    url = prefixes[i] + url.substring(prefixes[i].length)
+                }
+                break
+            }
+        }
+        if (!hasPrefix && prefixes.isNotEmpty()) {
+            url = prefixes[0] + url
+        }
+        return url
+    }
+
+    private fun gatherLinks(
+        links: ArrayList<LinkSpec>,
+        s: Spannable,
+        pattern: Pattern,
+        schemes: Array<String>,
+        matchFilter: Linkify.MatchFilter?,
+        transformFilter: Linkify.TransformFilter?,
+    ) {
+        val m = pattern.matcher(s)
+        while (m.find()) {
+            val start = m.start()
+            val end = m.end()
+            if (matchFilter == null || matchFilter.acceptMatch(s, start, end)) {
+                val url: String = makeUrl(m.group(0), schemes, m, transformFilter)
+                val spec = LinkSpec(url = url, start = start, end = end)
+                links.add(spec)
+            }
+        }
+    }
+
+    private fun pruneOverlaps(links: ArrayList<LinkSpec>, text: Spannable) {
+        // Append spans added by Markwon to remove any overlap.
+        val urlSpans: Array<URLSpan> = text.getSpans(0, text.length, URLSpan::class.java)
+        for (i in urlSpans.indices) {
```

Comment:
You can use e `map` in this case.

Reply:
I think `forEach` is better than map here.